test/unit/validate_test.py

```diff
@@ -0,0 +1,175 @@
+"""Unit tests for the functions in the pudl.validate module."""
+
+import io
+
+import numpy as np
+import pandas as pd
+import pytest
+
+import pudl.validate as pv
+
+
+@pytest.mark.parametrize(
+    "data,weights,quantile,expected",
+    [
+        # Basic test case - single non-zero weight
+        ([1, 2, 3, 4, 5], [0, 0, 0, 1, 0], 0.5, 4),
+        # Uniform weights - should behave like regular quantile
+        ([1, 2, 3, 4, 5], [1, 1, 1, 1, 1], 0.5, 3),
+        ([1, 2, 3, 4], [1, 1, 1, 1], 0.5, 2.5),
+        # Edge cases for quantiles
+        ([1, 2, 3, 4, 5], [1, 1, 1, 1, 1], 0.0, 1),
+        ([1, 2, 3, 4, 5], [1, 1, 1, 1, 1], 1.0, 5),
+        # Non-uniform weights
+        ([1, 2, 3], [1, 2, 1], 0.5, 2),
+        ([10, 20, 30], [1, 3, 1], 0.5, 20),
+        # Different quantiles with same data
+        ([1, 2, 3, 4, 5], [1, 1, 1, 1, 1], 0.25, 1.75),
+        ([1, 2, 3, 4, 5], [1, 1, 1, 1, 1], 0.75, 4.25),
```

Comment:
This is covered by "Edge cases" above since `weighted_quantiles` doesn't special-case 0 or 1 quantiles
```suggestion
```